man/man8/zpool-remove.8

```diff
@@ -52,10 +52,10 @@
 Removes the specified device from the pool.
 This command supports removing hot spare, cache, log, and both mirrored and
 non-redundant primary top-level vdevs, including dedup and special vdevs.
-When the primary pool storage includes a top-level raidz vdev only hot spare,
-cache, and log devices can be removed.
-Note that keys for all encrypted datasets must be loaded for top-level vdevs
-to be removed.
+.Pp
+Top-level vdevs can only be removed if the primary pool storage does not contain
+a top-level raidz vdev, all top-level vdevs have the same sector size, and the
+keys for all encrypted datasets are loaded.
```

Comment:
Thanks for mentioning the sector size restriction, however I think in the updated wording we should make it clear these restriction don't apply to spares, cache and log devices.  Only top-level vdevs which are part of the primary pool storage (singletons, mirrors, dedup, and special vdevs).  The pool also cannot contain either raidz or draid vdevs.

Reply:
Good point. I'll re-work it to be clearer.

Reply:
@behlendorf I pushed 9f93ea4 with what I think is clearer wording.